Feature/368 사용자 이력서 공개 여부 선택 기능 추가

MrMirror21 opened the pull request:
## Related issue 🛠
closed #368
## Work Description ✏️
<img width="390px" src="https://github.com/user-attachments/assets/6ad8f698-e43b-4ee0-8496-c6c8d7b0c54d" />

1.  이력서 공개 여부 선택 기능 구현
유학생이 자신의 이력서를 고용주에게 공개할지 설정할 수 있는 토글 기능 추가
이력서 관리 페이지에서 "Make Resume Public" 토글 버튼으로 공개/비공개 전환 가능

2.  UI 컴포넌트 개선
ToggleButton 공통 컴포넌트 분리 및 재사용성 확보
프로필 카드를 사용자 타입별로 분리 (ProfileCard vs ResumeProfileCard)
유학생(USER)과 고용주(OWNER) 별 UI 분기 처리

3. API 및 데이터 처리
7.23 (유학생) 이력서 공개 / 비공개 변경하기 patchResumePublic API 함수 구현
usePatchResumePublic Tanstack Query 훅 구현
이력서 조회 시 is_public 필드 포함하여 현재 상태 반영

4. 테스트 코드 작성
이력서 공개 여부 변경 기능에 대한 유닛 테스트 및 통합 테스트 추가
사용자 타입별 UI 렌더링 테스트
API 호출 및 상태 변경 테스트

## Uncompleted Tasks 😅
N/A

## To Reviewers 📢
### 주요 변경점
- 사용자 타입별 분기 처리: useUserStore의 account_type을 활용하여 유학생에게만 토글 버튼이 표시되도록 구현
- 상태 관리: 로컬 상태(toggleOn)와 서버 상태 동기화 방식 확인
- 테스트 패턴: 통합 테스트에서 실제 사용자 시나리오를 반영한 테스트 케이스 작성
### 핵심 변경 파일
- src/components/Common/ToggleButton.tsx - 새로 추가된 토글 컴포넌트
- src/components/ManageResume/ResumeProfileCard.tsx - 이력서 공개 토글 기능 포함
- src/api/resumes.ts & src/hooks/api/useResume.ts - 공개 여부 변경 API
- src/__tests__/integration/resume-public.test.tsx - 통합 테스트
### 🧪 테스트 실행
```bash
pnpm test src/__tests__/integration/resume-public.test.tsx
pnpm test src/hooks/api/useResume.test.ts
```

p.s 로딩 컴포넌트 렌더링 잘 되는데 왜 영상 촬영 시엔 저렇게 들어갔는지 모르겠네요..

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신규 기능**
    - 프로필 이미지를 표시하는 ProfileImage 컴포넌트와 토글 스위치를 제공하는 ToggleButton 컴포넌트가 추가되었습니다.
    - 이력서 프로필 카드에서 공개/비공개 상태를 토글할 수 있는 기능이 도입되었습니다.
    - 이력서 공개 상태를 서버에 반영하는 API 및 관련 React Query 훅이 추가되었습니다.
    - 하단 시트(BottomSheetLayout)에서 배경 클릭 시 닫기 기능(isExitable 옵션)이 추가되었습니다.

- **버그 수정**
    - 없음

- **테스트**
    - 이력서 공개/비공개 토글, API, 훅, 컴포넌트에 대한 통합 및 단위 테스트가 추가되었습니다.
    - 다양한 이력서 시나리오를 위한 목(mock) 데이터가 확장되었습니다.

- **문서화**
    - 상세 테스트 결과 및 커버리지 리포트 자동 생성 및 PR 코멘트 기능이 워크플로우에 추가되었습니다.

- **기타**
    - 테스트 및 디버깅 관련 npm 스크립트와 개발용 의존성이 추가 및 개선되었습니다.
    - .gitignore에 테스트 디버깅 파일 무시 규칙이 추가되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MrMirror21 committed:
✨ feat: 이력서 공개여부 조회 / 수정 관련 api 예시 함수 추가 #368

MrMirror21 committed:
✨ feat: 이력서 공개 여부 선택 토글 ui 추가 #368

MrMirror21 committed:
♻️ refactor: 프로필 카드 컴포넌트 분리 #368

MrMirror21 committed:
♻️ refactor: 토글 버튼 ui 공통 컴포넌트로 분리 #368

MrMirror21 committed:
✨ feat: 프로필 카드의 유학생/고용주 별 ui 분리 #368

MrMirror21 committed:
✨ feat: 바텀시트 모달 배경 클릭 시 이탈 가능 옵션 추가 #368

MrMirror21 committed:
Merge branch 'dev' of https://github.com/Team-inglo/Giggle-Web into featrue/368-사용자-이력서-공개-여부-선택-기능-추가

MrMirror21 committed:
🐛 fix: url -> 사용자 유형으로 번역 기준 변경 #368

MrMirror21 committed:
🔧 chore: 완료된 api 명세서에 맞춰 api 함수, 훅 변경 #368

MrMirror21 committed:
🔧 chore: 테스트 편의성을 위한 커맨드 수정 #363

MrMirror21 committed:
✅ test: 이력서 공개 여부 변경 유닛, 통합 테스트 추가 #368

MrMirror21 committed:
🔧 chore: api 명세와 타입 동기화 #368

MrMirror21 committed:
♻️ refactor: JSX 구문으로 변경 #368

MrMirror21 committed:
✅ test: PR 시 테스트 결과를 체크해 보여주는 봇 추가 #363

naarang commented on src/components/ManageResume/ResumeProfileCard.tsx:
```diff
@@ -11,6 +13,7 @@ type ResumeProfileCardProps = {
   main_address: string;
   phone: string;
   email: string;
+  isPublic: boolean;
```
is_public으로 main_address처럼 스네이크 케이스로 통일하는 것은 어떨까용?? 아니면 main_address를 mainAddress로요!

naarang commented on src/components/Common/ProfileImage.tsx:
```diff
@@ -0,0 +1,38 @@
+  className?: string;
+};
+
+const ProfileImage = ({
+  src,
+  alt,
+  size = 'md',
+  className = '',
+}: ProfileImageProps) => {
+  const sizeClasses = {
+    sm: 'w-11 h-11',
+    md: 'w-16 h-16',
+    lg: 'w-20 h-20',
+  };
```
오! 훨씬 깔끔하네요👍

MrMirror21 committed:
🐛 fix: github actions 워크플로우 에러 수정 #363

naarang commented on src/components/ManageResume/ResumeProfileCard.tsx:
```diff
@@ -93,6 +94,7 @@ const ResumeProfileCard = ({
   isPublic = true,
 }: ResumeProfileCardProps) => {
   const navigate = useNavigate();
+  const pathname = useLocation().pathname;
```
사소한 부분이지만ㅎㅎ `const { pathname } = useLocation();` 로 구조분해할당 써도 좋을 것 같아요!

MrMirror21 committed:
✅ test: Github check API 사용으로 변경 #363

naarang reviewed (approved):
테스트코드까지 작성해주셨네요👍👍 github actions도 잘 동작하면 확인하기 정말 편해질 것 같습니다!!

MrMirror21 committed:
✅ test: github pages 배포 기능 제거와 워크 플로우 상 예외 처리 추가 #363

MrMirror21 committed:
✅ test: 테스트 개수를 정확히 파악하도록 수정 #363

MrMirror21 committed:
🐛 fix: 워크플로우 중 테스트 결과 파싱 로직 수정 #363

MrMirror21 committed:
feat: GitHub Actions 테스트 리포트 워크플로우 개선 - CI 환경 대응

MrMirror21 committed:
🐛 fix: CI 환경 테스트 파싱 디버깅 - 전체 내용 출력 및 강제 파싱 추가

MrMirror21 committed:
🐛 fix: CI 환경 vitest 출력 불완전 문제 해결

MrMirror21 committed:
🐛 fix: 이전 워크플로우 출력 결과를 활용하도록 수정 #363

MrMirror21 committed:
🐛 fix: 정규식 패턴 문제 수정 #363

MrMirror21 committed:
♻️ refactor: 워크플로우 최적화 및 파싱 로직 개선 #363

- actions/cache 활용해 빌드 시간 단축
- npx vitest test --reporter json 옵션 사용해 json 형태의 결과를 받아 파싱하도록 변경
